### PR TITLE
Homework7: Aksana Batyrova

### PR DESCRIPTION
**Pull request by AksanaBatyrova**

### CODE REVIEW

File: homework_7.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+совпадения c их позициями в тайном числе (то есть количество коров)
+и сколько угадано вплоть до позиции в тайном числе (то есть
+количество быков). При игре против компьютера игрок вводит
+комбинации одну за другой, пока не отгадает всю последовательность.
+Ваша задача реализовать программу, против которой можно сыграть
+в "Быки и коровы"
+"""
+
+ANSWER = 2310
+first_guess = input('Make a guess: ')
```
Comment (Stml89) [inline homework_7.py:18]: user can enter not a unique number, string and 3 or 5 symbols

File: homework_7.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+противнику. Противник сообщает в ответ, сколько цифр угадано без
+совпадения c их позициями в тайном числе (то есть количество коров)
+и сколько угадано вплоть до позиции в тайном числе (то есть
+количество быков). При игре против компьютера игрок вводит
+комбинации одну за другой, пока не отгадает всю последовательность.
+Ваша задача реализовать программу, против которой можно сыграть
+в "Быки и коровы"
+"""
+
+ANSWER = 2310
```
Comment (Stml89) [inline homework_7.py:17]: is it possible to generate a secret number before each game?

File: homework_7.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            bulls += 1
+        elif guess_li[i] in answer_li:
+            cows += 1
+    print(f'{cows} cows, {bulls} bulls')
+    if bulls == 4:
+        return 'You win!'
+    if bulls != 4:
+        guess = input('Make another guess: ')
+        return bulls_and_cows_game(answer, guess)
+    return None
```
Comment (Stml89) [inline homework_7.py:38]: why do you want to return None?

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_7.py (hunk 3 above)
Comment (AksanaBatyrova) [reply homework_7.py]: it was the way to overcome this issue: https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/inconsistent-return-statements.html

File: homework_7.py (hunk 1 above)
Comment (AksanaBatyrova) [reply homework_7.py]: fixed in new commits

File: homework_7.py (hunk 2 above)
Comment (AksanaBatyrova) [reply homework_7.py]: fixed in new commits

File: homework_7.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        digit = random.randint(0, 9)
+        if digit not in number:
+            number.append(digit)
+    return number
+
+
+def make_a_guess():
+    """function asking user to enter their guess and checking it"""
+    try:
+        first_guess = int(input('Make a guess: '))
```
Comment (Stml89) [inline homework_7.py:33]: still possible to enter non unique number

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_7.py (hunk 4 above)
Comment (Stml89) [reply homework_7.py:33]: But if you scroll a bit you find line #63, no checks there

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_7.py (hunk 4 above)
Comment (AksanaBatyrova) [reply homework_7.py:33]: fixed

**Review by Stml89 — APPROVED** [review]

**Pull request merged**